src/control/server/storage/bdev/backend_class.go

```diff
@@ -0,0 +1,166 @@
+//
+// (C) Copyright 2018-2021 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package bdev
+
+import (
+	"bytes"
+	"os"
+	"path/filepath"
+	"syscall"
+	"text/template"
+
+	"github.com/dustin/go-humanize"
+	"github.com/pkg/errors"
+
+	"github.com/daos-stack/daos/src/control/common"
+	"github.com/daos-stack/daos/src/control/logging"
+	"github.com/daos-stack/daos/src/control/server/storage"
+)
+
+const (
+	clsNvmeTemplate = `[Nvme]
+{{ $host := .Hostname }}{{ range $i, $e := .DeviceList }}    TransportID "trtype:PCIe traddr:{{$e}}" Nvme_{{$host}}_{{$i}}
+{{ end }}    RetryCount 4
+    TimeoutUsec 0
+    ActionOnTimeout None
+    AdminPollRate 100000
+    HotplugEnable No
+    HotplugPollRate 0
+`
+	// device block size hardcoded to 4096
+	clsFileTemplate = `[AIO]
+{{ $host := .Hostname }}{{ range $i, $e := .DeviceList }}    AIO {{$e}} AIO_{{$host}}_{{$i}} 4096
+{{ end }}`
+	clsKdevTemplate = `[AIO]
+{{ $host := .Hostname }}{{ range $i, $e := .DeviceList }}    AIO {{$e}} AIO_{{$host}}_{{$i}}
+{{ end }}`
+	clsMallocTemplate = `[Malloc]
+    NumberOfLuns {{.DeviceCount}}
+    LunSizeInMB {{.DeviceFileSize}}000
+`
+	gbyte   = 1000000000
+	blkSize = 4096
+)
+
+func createEmptyFile(log logging.Logger, path string, size int64) error {
+	if !filepath.IsAbs(path) {
+		return errors.Errorf("expected absolute file path but got relative (%s)", path)
+	}
+	if size == 0 {
```

Comment:
Yeah. Usually we represent sizes with a uint64, and given that you're converting it in at least one place below it would probably be best to change the function signature to accept a uint64.